File: UI/wwwroot/Questionnaires/FrontModel/Cat_Valor_Preguntas.js
//@ts-check
import { EntityClass } from "../../WDevCore/WModules/EntityClass.js";
//@ts-ignore
import { Cat_Tipo_Preguntas } from './Cat_Tipo_Preguntas.js';
import { Resultados_Pregunta_Tests } from './Resultados_Pregunta_Tests.js';
class Cat_Valor_Preguntas extends EntityClass {
    /** @param {Partial<Cat_Valor_Preguntas>} [props] */
    constructor(props) {
        super(props, 'EntityQuestionnaires');
        for (const prop in props) {
            this[prop] = props[prop];
        };
        this.Id_tipo_pregunta = undefined;
    }
   /**@type {Number}*/ Id_valor_pregunta;
   /**@type {String}*/ Descripcion;
   /**@type {String}*/ Estado;
   /**@type {Number}*/ Valor;
   /**@type {Date}*/ Created_at;
   /**@type {Date}*/ Updated_at;
   /**@type {Cat_Tipo_Preguntas} ManyToOne*/ Cat_Tipo_Preguntas;
   /**@type {Array<Resultados_Pregunta_Tests>} OneToMany*/ Resultados_Pregunta_Tests;
}
export { Cat_Valor_Preguntas };

File: BusinessLogic/Questionnaires/Mapping/Model/Cat_Valor_Preguntas.cs
using CAPA_DATOS;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
namespace DataBaseModel {
   public class Cat_Valor_Preguntas : EntityClass {
       [PrimaryKey(Identity = true)]
       public int? Id_valor_pregunta { get; set; }
       public string? Descripcion { get; set; }
       public int? Id_tipo_pregunta { get; set; }
       public string? Estado { get; set; }
       public int? Valor { get; set; }
       public DateTime? Created_at { get; set; }
       public DateTime? Updated_at { get; set; }
       [ManyToOne(TableName = "Cat_Tipo_Preguntas", KeyColumn = "Id_tipo_pregunta", ForeignKeyColumn = "Id_tipo_pregunta")]
       public Cat_Tipo_Preguntas? Cat_Tipo_Preguntas { get; set; }
       [OneToMany(TableName = "Resultados_Pregunta_Tests", KeyColumn = "Id_valor_pregunta", ForeignKeyColumn = "Id_valor_pregunta")]
       public List<Resultados_Pregunta_Tests>? Resultados_Pregunta_Tests { get; set; }
   }
}
